subsys/mgmt/mcumgr/grp/img_mgmt/Kconfig

```diff
@@ -61,6 +61,38 @@ config MCUMGR_GRP_IMG_VERBOSE_ERR
 	  Add additional "rsn" key to SMP responses, where provided, explaining
 	  non-0 "rc" codes.
 
+config MCUMGR_GRP_IMG_ALLOW_CONFIRM_NON_ACTIVE_IMAGE_SECONDARY
+	bool "Allow to confirm secondary slot of non-active image"
+	default y
+	help
+	  Allows to confirm secondary (non-active) slot of non-active image.
+	  Normally it should not be allowed to confirm any slots of non-active
+	  image, via MCUmgr commands, to prevent confirming something that is
+	  broken and may not boot in other slot; instead application should
+	  have means to test and confirm the image.
+	  Unfortunately due to logic bug it has been allowed to confirm
+	  secondary slot of non-active image so this option is set to y
+	  by default.
```

Comment:
I do not agree with this comment, it is not a bug, users are free to confirm any images they want, even if it then ends up being a broken image. And for mcuboot in upgrade only mode, confirmation is required because there is no reverting

Reply:
Should I remove the Kconfigs that allow to confirm SECONDARY/ANY?

Reply:
The comment has been addressed in base PR.